apps/api/src/test-data/utilities/convertInMemorySnapshotToDatabaseFormat.ts

```diff
@@ -0,0 +1,57 @@
+import BaseDomainModel from '../../domain/models/BaseDomainModel';
+import { AggregateId } from '../../domain/types/AggregateId';
+import {
+    InMemorySnapshot,
+    InMemorySnapshotOfResources,
+    ResourceType,
+} from '../../domain/types/ResourceType';
+import { UuidDocument } from '../../lib/id-generation/types/UuidDocument';
+import { ArangoCollectionId } from '../../persistence/database/collection-references/ArangoCollectionId';
+import { ArangoEdgeCollectionId } from '../../persistence/database/collection-references/ArangoEdgeCollectionId';
+import { getArangoCollectionIDFromResourceType } from '../../persistence/database/collection-references/getArangoCollectionIDFromResourceType';
+import buildEdgeDocumentsFromCategoryNodeDTOs from '../../persistence/database/utilities/category/buildEdgeDocumentsFromCategoryNodeDTOs';
+import mapCategoryDTOToArangoDocument from '../../persistence/database/utilities/category/mapCategoryDTOToArangoDocument';
+import mapEdgeConnectionDTOToArangoEdgeDocument from '../../persistence/database/utilities/mapEdgeConnectionDTOToArangoEdgeDocument';
+import mapEntityDTOToDatabaseDTO from '../../persistence/database/utilities/mapEntityDTOToDatabaseDTO';
+import { DTO } from '../../types/DTO';
+
+type InMemoryDatabaseSnapshot = {
+    documentCollections: {
+        [K in Exclude<ArangoCollectionId, 'uuids'>]: unknown[];
+    } & { uuids: UuidDocument<AggregateId>[] };
+
+    edgeCollections: {
+        [K in ArangoEdgeCollectionId]: Record<string, unknown>[];
+    };
+};
+
+const toDto = <T extends BaseDomainModel>(model: T): DTO<T> => model.toDTO();
+
+export default (snapshot: InMemorySnapshot): InMemoryDatabaseSnapshot => {
```

Comment:
Now might be a good time to explore turning the `InMemorySnapshot` into a proper data structure with methods. E.g. a tree which allows you to map, filter, and reduce by applying callbacks to the leaves.

Reply:
Story [here](https://www.pivotaltracker.com/story/show/182693855).